### PR TITLE
feat(core): submit UX improvements

### PR DESCRIPTION
**Pull request by rdunnington**
* RepoStatus now stores lock information
* Removed verify_locks calls from Friendshipper in favor of RepoStatus locks
* RepoStatus Files now have the lock owner, flags for submit status, and git state (added/ modified/etc)
* Friendshipper submit UX has been improved to display if files can be submitted or not, colorizing them and displaying a warning icon if there will be a problem with submission.

### CODE REVIEW

**Comment by rdunnington** [comment]
Also closes #78

File: friendshipper/src-tauri/src/engine/unreal/ofpa.rs [hunk 1]
```diff
@@ -160,12 +160,13 @@ impl OFPANameCache {
...
             for path in paths {
                 let neeeds_translate = Self::path_needs_translate(path);
+                let can_request = communication != CommunicationType::None;
                 // If the editor is running, the user could have changed the name of the asset, so attempt to fetch an updated
                 // name for it if a refresh is desired. If the editor is closed, the last name we have in the cache is probably
                 //  good enough since we run status updates pretty often.
                 let needs_request = is_editor_running || !cache.path_to_name_map.contains_key(path);
 
-                if neeeds_translate && needs_request {
+                if neeeds_translate && can_request && needs_request {
```
Comment (rudoi) [inline friendshipper/src-tauri/src/engine/unreal/ofpa.rs:169]: `neeeeeeeeeeeeeeeeeeeeeeeeeeds`

Comment (rdunnington) [reply friendshipper/src-tauri/src/engine/unreal/ofpa.rs:169]: haha

Comment (rdunnington) [reply friendshipper/src-tauri/src/engine/unreal/ofpa.rs:169]: I'll merge and fix in an update

Comment (rdunnington) [reply friendshipper/src-tauri/src/engine/unreal/ofpa.rs:169]: mind if I fix this in a followup change?

**Review by rudoi — APPROVED** [review]

**Pull request merged**